### PR TITLE
fix(cluster) quiet keepalive events

### PR DESCRIPTION
**Pull request by thefosk**
### Full changelog

* Stops propagating cluster keep-alive requests, reducing intracluster noise

### CODE REVIEW

File: kong/dao/dao.lua [hunk 1]
```diff
@@ -102,11 +102,12 @@ end
 --- Insert a row.
 -- Insert a given Lua table as a row in the related table.
 -- @param[type=table] tbl Table to insert as a row.
--- @param[type=table] options Options to use for this insertion. (`ttl`: Time-to-live for this row, in seconds)
+-- @param[type=table] options Options to use for this insertion. (`ttl`: Time-to-live for this row, in seconds, `quiet`: does not send event)
 -- @treturn table res A table representing the insert row (with fields created during the insertion).
 -- @treturn table err If an error occured, a table describing the issue.
 function DAO:insert(tbl, options)
   check_arg(tbl, 1, "table")
+  options = options or {}
```
Comment (thibaultcha) [inline kong/dao/dao.lua]: Should be: 
```lua
function DAO:insert(tbl, options)
  options = options or {}
  check_arg(tbl, 1, "table")		    
  check_arg(options, 2, "table")
 ```

File: kong/core/cluster.lua [hunk 2]
```diff
@@ -82,7 +82,7 @@ local function send_keepalive(premature)
       ngx_log(ngx.ERR, tostring(err))
     elseif #nodes == 1 then
       local node = nodes[1]
-      local _, err = singletons.dao.nodes:update(node, node, {ttl=singletons.configuration.cluster_ttl_on_failure})
+      local _, err = singletons.dao.nodes:update(node, node, {ttl=singletons.configuration.cluster_ttl_on_failure, quiet = true})
```
Comment (thibaultcha) [inline kong/core/cluster.lua]: Could we avoid long lines? As per our codestyle, not exceeding 80 chars width is ideal:
```lua
local _, err = singletons.dao.nodes:update(node, node, {
  ttl = singletons.configuration.cluster_ttl_on_failure, 
  quiet = true
})
```

File: kong/dao/dao.lua (hunk 1 above)
Comment (thefosk) [inline kong/dao/dao.lua]: Done

File: kong/core/cluster.lua (hunk 2 above)
Comment (thefosk) [inline kong/core/cluster.lua]: Done

**Comment by Tieske** [comment]
two questions;

1. no tests?
1. shouldn't, for consistency, the `quiet` option be implemented for `DELETE` as well?

**Comment by thibaultcha** [comment]
> no tests?

Not testable, at least none are written regarding events propagation (other than invalidation), and as we know they are *extremely* unreliable. Testing that an event was *not* propagated is even trickier. 

> shouldn't, for consistency, the quiet option be implemented for DELETE as well?

Eventually. Before the end of the day if so, because `0.9.2` is to be released tomorrow. Otherwise will merge this as-is.

**Comment by thibaultcha** [comment]
Has been implemented in `delete` as well.

**Comment by thefosk** [comment]
I was working on the tests

**Pull request merged**